docs/source/frameworks/fastai.rst

```diff
@@ -0,0 +1,246 @@
+
+.. note::
+
+   By default, BentoML uses ``cloudpickle`` for learner serialization. The key difference from Python's ``pickle`` is that 
+   ``cloudpickle`` has the capability to serialize functions and so it can directly serialize members of the object without reference to its type.
+
+   If the :func:`save_model` method failed while saving a given learner, your learner may contain a :obj:`~fastai.callback.core.Callback` that is not picklable.
+   All FastAI callbacks are stateful, which makes some of them not picklable. Use :func:`Learner.remove_cbs` to remove unpicklable callbacks.
+
+.. tip::
+
+   We found that some of given callback are only useful during training phase, not during serving. Therefore, removing callback will not affect inference results.
+
+
+Loading a learner with BentoML
+------------------------------
+
+.. warning::
+
+   We recommend users to to use ``load_model`` inside a :obj:`bentoml.Service`.
+
+   You should always use ``bentoml.models.get("model:tag").to_runner()`` to get
+   a :obj:`bentoml.Runner` instead. See also :ref:`Runners <concepts/runner:Using Runners>`_ for more information.
+
+
+To load the learner back to memory, use ``load_model``:
+
+.. code-block:: python
+
+   learner = bentoml.fastai.load_model("fastai_sentiment:latest")
+
+You can then proceed to test the learner with prediction inputs:
+
+.. code-block:: python
+
+   learner.predict("I really liked that movie!")
+
+.. admonition:: About the behaviour of :code:`load_model()`
+
+   Since fastai doesn't provide a good support for GPU during inference, BentoML
+   by default will only support CPU inference for fastai. If you want to use
+   GPU, you should get the ``PyTorch`` model from ``learner.model`` and then use
+   ``bentoml.pytorch`` instead.
+
+   Additionally, if the model uses ``mixed_precision``, then the loaded model will also be converted to FP32.
+   See `mixed precision <https://docs.fast.ai/callback.fp16.html>`_ to learn more about mixed precision.
+
+
+Using Runners
+-------------
+
+.. seealso::
+
+   :ref:`The general Runner documentation<concepts/runner:Using Runners>`: general information about the Runner concept and their usage.
+
+.. seealso::
+
+   :ref:`Specifying Runner Resources<concepts/runner:Specifying Required Resources>`: more information about Runner options.
+
+
+To use fastai runner locally, access the model via ``get`` and convert it to
+a runner:
+
+.. code-block:: python
+
+   runner = bentoml.fastai.get("fastai_sentiment").to_runner()
+
+   runner.init_local()
+
+   runner.predict.run("I really liked that movie!")
+
+.. tip::
+
+   ``runner.predict.run`` should generally be a drop-in replacement for ``learner.predict`` regardless of your learner type.
+
+.. admonition:: About adaptive batching in fastai
+
+   fastai doesn't have support for multiple inputs, hence adaptive batching
```

Comment:
By "multiple inputs" do we mean "batched inputs", because "multiple inputs" can be interpreted as multiple input types, e.g. `input=(metadata=JSON(), image=Image())`.

Reply:
fastai does not support batched inputs and multiple input types.

Reply:
multiple input -> batched input?